### ISSUE TITLE
ESXi Hosts without a VMware Cluster fail to be added

### ISSUE THREAD

**Issue by fraserc182** [synackray/vcenter-netbox-sync#63]
### Environment
* Python version:  3.6.7
* NetBox version: 2.6.8
* vCenter version: 6.5.0
* vCenter NetBox Sync version:  

### Steps to Reproduce
1. Follow steps in readme.md and install.
2. Execute run.py -v

### Expected Behavior
For devices to be created in netbox

### Observed Behavior

I firstly got an error message regarding null error message, I got round this with the workaround in this issue: https://github.com/synackray/vcenter-netbox-sync/issues/58
This is odd because I have the most up to date build as I only cloned it a few hours ago.

Anyway I now have an error that says:  

```
NetBox 400 status reason: {'cluster': ["Related object not found using the provided attributes: {'name': 'edimandvesxh001.lambienet.local'}"]}
```

This then causes all the other devices to error because the device it is trying to add them to does not exist.
Unsure why it cannot get a serial number for this device.
Full debug is attached.
[output.txt](https://github.com/synackray/vcenter-netbox-sync/files/4180577/output.txt)

**Comment by synackray** [comment]
Hi @fraserc182! I'll look into this immediately.

`This is odd because I have the most up to date build as I only cloned it a few hours ago.`
Are you running the latest commit from the `develop` branch or the last release on `master`?

**Comment by fraserc182** [comment]
Hi,

The last release on master! I can grab the develop branch if this is where the serial address fix is.

As for the other error:

```
NetBox 400 status reason: {'cluster': ["Related object not found using the provided attributes: {'name': 'edimandvesxh001.lambienet.local'}"]}
```

I manually created a cluster named edimandvesxh001.lambienet.local in netbox and then it ran without any problems. So it looks like it is having issues with creating this cluster? 
Thanks for getting back to me so fast.

**Comment by synackray** [comment]
`I cloned the development build and this resolves the serial number issue.`

Thanks for the update. I am planning to promote to `master` soon, pending final regression testing. Are there any outstanding issues in your sync with `develop`?

**Comment by fraserc182** [comment]
Yes I am still getting this error:

```
NetBox 400 status reason: {'cluster': ["Related object not found using the provided attributes: {'name': 'edimandvesxh001.lambienet.local'}"]}
```

Manually creating the cluster it's looking for fixes it but I feel I'm sidestepping something by doing this.

**Comment by synackray** [comment]
Hmm something doesn't look right with the way the cluster is being parsed after collection. I am sorry to ask, but I may need to have you run a special branch to get additional detail. First I'll try to recreate the issue on my side. 

```
2020-02-10 11:43:25,948 [INFO] Initiated sync of vCenter cluster objects to NetBox.
2020-02-10 11:43:25,948 [INFO] Collecting vCenter cluster objects.
2020-02-10 11:43:25,967 [DEBUG] Collected 1 vCenter cluster object.
2020-02-10 11:43:25,967 [INFO] Starting sync of 0 vCenter clusters objects to NetBox clusters objects.
2020-02-10 11:43:25,968 [INFO] Finished sync of 0 vCenter clusters objects to NetBox clusters objects.
```

**Comment by fraserc182** [comment]
Sure thing, just let me know what you need from me.

**Comment by synackray** [comment]
I suspect this may related to the structure of your particular vCenter resources. Is the host `edimandvesxh001.lambienet.local` actually assigned to a cluster or is it standalone and perhaps in a folder instead? My reasoning for this is that the cluster objects call on their parent resources as shown in the screenshot below. 

![image](https://user-images.githubusercontent.com/24757919/74157045-de8d9180-4c17-11ea-9ac5-d5a25b1c566a.png)
 
If a cluster is not created then it would seem no clusters would be returned and perhaps vCenter returns the host itself as the parent object.. I've never run into the situation but it's a theory. I'll try to spin this up in lab and see if I can reproduce.

**Comment by fraserc182** [comment]
This is our lab environment so it's just standalone.
It is just a single esxi host on bare metal, so you're right it's not part of a cluster!
I suppose this is expected behaviour then.
Maybe I can manually name the cluster instead of it getting it from the parent object?

**Comment by synackray** [comment]
I see. I've never actually tested the script on a single host. You should be able to create the cluster object, the rules for it are very simple. The platform is "VMware ESXi", the default site is "vCenter" and the device role is "Server". 

This is an interesting use case though. I'll do some testing on this and see if it's feasible for future support.

**Comment by fraserc182** [comment]
Brilliant, thanks for looking into this for me so quickly!
I'll just add the cluster manually.

Cheers for creating this project!

**Comment by synackray** [comment]
I've added the ability to import hosts that are not part of parent clusters while patching #82. Thanks for your report it was a big help.

**Comment by Haeki** [comment]
Hi,
i had the same issue with datacenters with only one Host not being added properly to Netbox. But I found a different solution: I changed the `vc_obj_views` value for clusters from `vim.ClusterComputeResource` to `vim.ComputeResource` this means datacenters with a single Hosts have a Cluster added to Netbox with the host name as the cluster name. This cluster also gets the correct cluster group and host device added.

I am interested to hear feedback on my approach and what disadvantages it may cause. But on first look it seems a lot simpler because the it is only a one line change and the rest is handled by the existing script.
